src/text_snippets.cpp

```diff
@@ -246,6 +246,12 @@ std::string snippet_library::expand( const std::string &str ) const
     std::string symbol = str.substr( tag_begin, tag_end - tag_begin + 1 );
     std::optional<translation> replacement = random_from_category( symbol );
     if( !replacement.has_value() ) {
+        if( with_tags ) {
```

Comment:
This flag and the recursive calls between `parse_basic_tags` and `expand` feel ugly to me. It's probably better to always call `parse_static_tags` from `expand` and always expand the tags using `expand` elsewhere.